config.go

```diff
@@ -27,11 +28,48 @@ type repository struct {
 func (r repository) Packages() []string {
 	pkgs := []string{r.Prefix}
 	for _, s := range r.Subs {
-		pkgs = append(pkgs, path.Join(r.Prefix, s))
+		pkgs = append(pkgs, path.Join(r.Prefix, s.Name))
 	}
 	return pkgs
 }
 
+type sub struct {
+	Name   string `json:"name"`
+	Hidden bool   `json:"hidden"`
+}
+
+func (s *sub) UnmarshalJSON(raw []byte) error {
+	var v interface{}
+
+	err := json.Unmarshal(raw, &v)
+	if err != nil {
+		return err
+	}
+
+	switch t := v.(type) {
+	case string:
+		s.Name = t
+
+	case map[string]interface{}:
+		var rs struct {
+			Name   string `json:"name"`
+			Hidden bool   `json:"hidden"`
+		}
+		err := json.Unmarshal(raw, &rs)
+		if err != nil {
+			return err
+		}
+
+		s.Name = rs.Name
+		s.Hidden = rs.Hidden
+
+	default:
+		return errors.New("unknown sub type")
```

Comment:
Can we make this error message more like the errors that the json module outputs when it has encountered an incompatible type? e.g.:

>cannot unmarshal object into Go struct of type [type]

Where `[type]` is the type that has been deserialized.